data2bids.m

```diff
@@ -1536,8 +1536,16 @@
     % use the events table as it is
     events_tsv = cfg.events;
   elseif istable(cfg.events) && all(ismember({'begsample', 'endsample', 'offset'}, fieldnames(cfg.events)))
-    % it is a "trl" matrix formatted as table, use it as it is
+    % it is a "trl" matrix formatted as table, use it as it is, but add
+    % onset and duration
     events_tsv = cfg.events;
+    begsample                   = table2array(events_tsv(:,{'begsample'}));
+    endsample                   = table2array(events_tsv(:,{'endsample'}));
+    onset                       = begsample./hdr.Fs;
```

Comment:
the first sample of the recording is 1, but the first time in the recording is 0. So `onset = (begsample-1)/hdr.Fs`

Reply:
Yes, my mistake!

Reply:
but as far as I can see, you have not changed the code yet